File: solutions/cpp/grains/1/grains.cpp
#include "grains.h"

#include <cmath>

namespace grains
{
    unsigned long long square(int n)
    {
        return pow(2.0L, n - 1);
    }

    unsigned long long total()
    {
        return pow(2.0L, 64) - 1;
    }
} // namespace grains

File: solutions/python/grains/1/grains.py
"""
Calculate the number of grains of wheat on a chessboard given that the number
on each square doubles.
"""

def square(number):
    """
    :param number: int - square number
    :return: int - how many grains were on a given square
    """
    if number < 1 or number > 64:
        raise ValueError('square must be between 1 and 64')

    return 2**(number-1)


def total():
    """
    :return: int - the total number of grains on the chessboard
    """
    return 2**64-1
